### PR TITLE
added logger

### PR DESCRIPTION
**Pull request by andriylagun**

### CODE REVIEW

**Review by YaTut1901 — APPROVED** [review]

File: src/main/java/com/online/cinema/dao/impl/GenericDaoImpl.java [hunk 1]
```diff
@@ -50,12 +55,14 @@ public T add(T element) {
...
 
     protected T get(Class<T> clazz, Long id) {
+        logger.info("Trying to get entity");
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
             return session.get(clazz, id);
         }
     }
 
     public void update(T entity) {
+        logger.info("Trying to update entity");
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/cinema/dao/impl/GenericDaoImpl.java:65]: let's add a log if everything went ok here

File: src/main/java/com/online/cinema/dao/impl/GenericDaoImpl.java [hunk 2]
```diff
@@ -5,11 +5,13 @@
 import com.online.cinema.util.HibernateUtil;
 import java.util.List;
 import javax.persistence.criteria.CriteriaQuery;
+import org.apache.log4j.Logger;
 import org.hibernate.Session;
 import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
 
 public abstract class GenericDaoImpl<T> implements GenericDao<T> {
+    protected static Logger logger = Logger.getLogger(GenericDaoImpl.class);
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/cinema/dao/impl/GenericDaoImpl.java]:
```suggestion
    private static final Logger logger = Logger.getLogger(GenericDaoImpl.class);
```

File: src/main/java/com/online/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 3]
```diff
@@ -31,7 +31,9 @@ public MovieSession get(Long id) {
 
     @Override
     public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            logger.info("Trying to get available sessions");
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/cinema/dao/impl/MovieSessionDaoImpl.java:38]: it is better to have logger object in each of the classes
for logs to be more informative

File: src/main/java/com/online/cinema/Main.java [hunk 4]
```diff
@@ -41,7 +43,7 @@ public static void main(String[] args) {
         CinemaHall cinemaHall2 = new CinemaHall();
         cinemaHall2.setDescription("HALL NUMBER TWO");
         cinemaHallService.add(cinemaHall2);
-        cinemaHallService.getAll().forEach(System.out::println);
+        cinemaHallService.getAll().forEach(logger::info);
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/cinema/Main.java:46]: there is still one more sout above

File: src/main/java/com/online/cinema/Main.java [hunk 5]
```diff
@@ -70,39 +72,39 @@ public static void main(String[] args) {
...
         authenticationService.register("user2@gmail.com", "pass");
         User user = userService.findByEmail("user1@gmail.com").get();
-        System.out.println(user.toString());
+        logger.info(user.toString());
         try {
-            System.out.println(authenticationService.login("user2@gmail.com", "pass").toString());
+            logger.info(authenticationService.login("user2@gmail.com", "pass").toString());
         } catch (AuthenticationException e) {
-            e.printStackTrace();
+            logger.error(e.getMessage(), e);
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/cinema/Main.java]:
```suggestion
            logger.warn("Login failed: " + e.getMessage(), e);
```
nothing really bad happened here, `warn` level is enough

File: src/main/java/com/online/cinema/Main.java [hunk 6]
```diff
@@ -70,39 +72,39 @@ public static void main(String[] args) {
...
         }
 
         ShoppingCartService shoppingCartService
                 = (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
         shoppingCartService.addSession(movieSession1, user);
         shoppingCartService.addSession(movieSession2, user);
-        System.out.println(shoppingCartService.getByUser(user));
+        logger.info(shoppingCartService.getByUser(user));
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/cinema/Main.java]: it would be nice to add some informative messages to your logs

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/online/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 7]
```diff
@@ -13,11 +13,13 @@
 import javax.persistence.criteria.JoinType;
 import javax.persistence.criteria.Predicate;
 import javax.persistence.criteria.Root;
+import org.apache.log4j.Logger;
 import org.hibernate.Session;
 
 @Dao
 public class MovieSessionDaoImpl extends GenericDaoImpl<MovieSession>
         implements MovieSessionDao {
+    private static Logger logger = Logger.getLogger(MovieSessionDaoImpl.class);
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/cinema/dao/impl/MovieSessionDaoImpl.java]:
```suggestion
    private static final Logger logger = Logger.getLogger(MovieSessionDaoImpl.class);
```

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/com/online/cinema/dao/impl/MovieSessionDaoImpl.java (hunk 7 above)
Comment (andriylagun) [reply src/main/java/com/online/cinema/dao/impl/MovieSessionDaoImpl.java]: missed)

**Pull request merged**